### DISCUSSION TITLE
Quick question,does clangd not support c++

### DISCUSSION THREAD

**Discussion by RESDXChgfore9hing** [clangd/clangd#1418] [General]
It seems like the include doesnt *include the c++ std lib headers and include files
such as iostream
Thus when using it in lsp for c++ it gives error.

**Comment by HighCommander4** [comment]
Please see https://clangd.llvm.org/troubleshooting#cant-find-standard-library-headers-map-stdioh-etc

**Reply by RESDXChgfore9hing** [reply]
i checked https://clangd.llvm.org/troubleshooting#cant-find-standard-library-headers-map-stdioh-etc

it seems to mention the need to setup some configuration file to point to the compiler is it?
Is it .clangd or compile_commands.json
and if so what do i type inside?Is there a template?

**Reply by HighCommander4** [reply]
The first step is for the project to have a `compile_commands.json` file. It's typically generated by a tool, not written manually. Please review https://clangd.llvm.org/installation#project-setup for ways to generate it.

Once you have a `compile_commands.json`, the way to resolve issues around clangd not being able to find standard library headers is usually to add `--query-driver=<path>` to the clangd command line, where `<path>` is the full path to your compiler that also appears at the beginning of the commands in `compile_commands.json`. Clangd command-line arguments are specified in your editor's settings, see https://clangd.llvm.org/installation#editor-plugins for more details.
